packages/web-components/fast-foundation/src/combobox/combobox.ts

```diff
@@ -399,6 +402,11 @@ export class Combobox extends FormAssociatedCombobox {
         this.filter = this.control.value;
         this.filterOptions();
 
+        if (this.filteredOptions.length == 0) {
+            // if no matching value deselect any selected item in popup
+            this.selectedIndex = -1;
+        }
```

Comment:
In `this.filterOptions()`, `this.filter` is set to `""` if `this.autocomplete` is undefined or `none`. That means that this would always reset the `selectedIndex` to -1 on input, even if the current value matches the text of an available option.

Reply:
Note that the `selectedIndex` will get updated appropriately upon commit. As is, I believe the behavior matches what is there currently (when `autocomplete` is set to `none`). However, I could see it being nice to update the `selectedIndex` to the appropriate item whilst typing (again, when `autocomplete` is `none`). Would you want me to do this, or would you rather just introduce a guard to prevent us setting `selectedIndex` to -1 if there is no autocomplete.

Reply:
@radium-v, as you can see, I am now setting `selectedIndex` to whatever the exact match is for the non-inline autocomplete cases. I've added tests to reflect this change.